index.js

```diff
@@ -103,7 +81,7 @@ var run = function(options) {
     };
 
     if (options.publish)
-      publish(name, output, extension, options, written);
+      S3Publisher.publish(output, { format: extension.slice(1) })
```

Comment:
Coming back to this. Looks like I forgot to call `written` once the publisher resolves.